datagateway_api/src/search_api/helpers.py

```diff
@@ -14,44 +16,154 @@
 
 
 @client_manager
-def get_search(endpoint_name, entity_name, filters):
+def get_search(entity_name, filters):
+    """
+    Search for data on the given entity, using filters from the request to restrict the
+    query
+
+    :param entity_name: Name of the entity requested to query against
+    :type entity_name: :class:`str`
+    :param filters: The list of Search API filters to be applied to the request/query
+    :type filters: List of specific implementation :class:`QueryFilter`
+    :return: List of records (in JSON serialisable format) of the given entity for the
+        query constructed from that and the request's filters
+    """
+
+    log.info("Searching for %s using request's filters", entity_name)
     log.debug("Entity Name: %s, Filters: %s", entity_name, filters)
 
-    icat_relations = mappings.get_icat_relations_for_panosc_non_related_fields(
-        entity_name,
-    )
-    # Remove any duplicate ICAT relations
-    icat_relations = list(dict.fromkeys(icat_relations))
-    if icat_relations:
-        filters.append(PythonICATIncludeFilter(icat_relations))
-
     query = SearchAPIQuery(entity_name)
 
     filter_handler = FilterOrderHandler()
     filter_handler.add_filters(filters)
+    icat_relations = filter_handler.add_icat_relations_for_panosc_non_related_fields(
+        entity_name,
+    )
```

Comment:
```suggestion
    filter_handler.add_icat_relations_for_panosc_non_related_fields(entity_name)
```
Because `add_icat_relations_for_panosc_non_related_fields` method will be void

Reply:
Fixed in 4343a2c67cdba63a0b21e552b7d630eaf0b46cd6